src/components/widget/widget.directive.coffee

```diff
@@ -53,19 +53,22 @@ module.controller('ImpacWidgetCtrl', ($scope, $log, $q, $timeout, ImpacWidgetsSv
     return classes.join(' ')
 
   ImpacDashboardsSvc.pdfModeEnabled().then(null, null, ->
-    # if w.expanded is defined, the width directive will handle the display
-    unless angular.isDefined(w.expanded)
+    # Save whether the widget is expanded
```

Comment:
Shouldn't we should move the logic related to the width management in the width directive?
Maybe something like:
```coffeescript
# in widget directive
ImpacDashboardsSvc.pdfModeEnabled().then(null, null, ->
  $scope.pdfMode = true
  w.ticked = true unless angular.isDefined(w.ticked)
  $scope.widget.hasEditAbility = false

# in widget directive
ImpacDashboardsSvc.pdfModeCanceled().then(null, null, ->
  $scope.pdfMode = false
  $scope.widget.hasEditAbility = true

# ----------------------------------

# in width directive
ImpacDashboardsSvc.pdfModeEnabled().then(null, null, ->
  $scope.pdfMode = true
  if w.isExpanded()      # should always be defined in this case
    $scope.initiallyExpanded = true
  else
    $scope.initialWidth = w.width
    w.width = 12
    # I don't think the below will still be necessary?
    # $scope.initSettings()
  w.toggleExpanded(false) unless $scope.expanded

# in width directive
ImpacDashboardsSvc.pdfModeCanceled().then(null, null, ->
  $scope.pdfMode = false
  unless w.isExpanded()      # should always be defined in this case
    w.width = $scope.initialWidth
  w.toggleExpanded(false) unless $scope.initiallyExpanded
```

Imho it would be more clear as the concerns would be separated between what regards the width and the widget in itself, + you would minimise the use of variables attached to `w`...

By looking at it closer, I realise there is something very weird with the width management: have a look at the `toggleExpanded` function (in width directive): it looks like we are modifying the width again there... You can probably also wrap some code of the sample above in the `toggleExpanded` function and reduce the number of variables / methods used...

What do you think?